lib/castle/rollup_log.ex

```diff
@@ -2,51 +2,70 @@ defmodule Castle.RollupLog do
   use Ecto.Schema
   import Ecto.Changeset
 
-  @beginning_of_time "2017-04-07"
   @buffer_seconds 300
+  @beginning_of_time ~D[2017-04-01]
+  def beginning_of_time, do: @beginning_of_time
 
   schema "rollup_logs" do
     field :table_name, :string
     field :date, :date
+    field :complete, :boolean, default: false
     timestamps()
   end
 
   def changeset(rollup_log, attrs) do
     rollup_log
-    |> cast(attrs, [:table_name, :date])
-    |> validate_required([:table_name, :date])
+    |> cast(attrs, [:table_name, :date, :complete])
+    |> validate_required([:table_name, :date, :complete])
   end
 
-  def upsert(log) do
-    conflict = [set: [updated_at: Timex.now()]]
+  def upsert!(log) do
+    conflict = [set: [updated_at: Timex.now(), complete: log.complete]]
     target = [:table_name, :date]
     Castle.Repo.insert!(log, on_conflict: conflict, conflict_target: target)
   end
 
-  def find_missing(table_name, limit, to_date \\ nil) do
+  def find_missing_days(tbl, lim), do: find_missing_days(tbl, lim, default_to_date())
+  def find_missing_days(table_name, limit, to_date) do
+    find_missing table_name, limit, """
+      SELECT r.DATE as date
+      FROM GENERATE_SERIES('#{date_str(to_date)}', '#{@beginning_of_time}', '-1 DAY'::INTERVAL) r
+    """
+  end
+
+  def find_missing_months(tbl, lim), do: find_missing_months(tbl, lim, default_to_date())
+  def find_missing_months(table_name, limit, to_date) do
```

Comment:
All the other rollups get 1-row-per-day ... but MonthlyDownloads only have a row for the 1st of every month.